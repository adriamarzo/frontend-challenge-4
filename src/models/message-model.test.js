import Message from './message-model';

const DATA_MOCK = {
  key: 'key',
  title: 'The title',
  samurai: true,
  timestamp: 1637533924211,
  type: 'text',
};

describe('Message model', () => {
  it('should have the expected properties', () => {
    const message = new Message(DATA_MOCK);

    expect(message.toJSON()).toMatchSnapshot();
  });

  it('should have the text', () => {
    const message = new Message({
      ...DATA_MOCK,
      message: DATA_MOCK.title,
      title: null,
    });

    expect(message.toJSON().text).toBe(DATA_MOCK.title);
  });

  it('should be a user message', () => {
    const message = new Message({
      ...DATA_MOCK,
      samurai: undefined,
    });

    expect(message.toJSON().author).toBe('user');
  });
});
