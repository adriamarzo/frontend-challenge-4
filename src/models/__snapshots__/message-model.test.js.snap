// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message model should have the expected properties 1`] = `
Object {
  "author": "bot",
  "key": "key",
  "text": "The title",
  "timestamp": 1637533924211,
  "type": "text",
}
`;
